### PR TITLE
ratelimit incoming RPCs

### PR DESCRIPTION
**Pull request by DavidVorick**

### CODE REVIEW

File: modules/gateway/consts.go [hunk 1]
```diff
@@ -73,6 +73,15 @@ var (
 		Testing:  500 * time.Millisecond,
 	}).(time.Duration)
 
+	// peerRPCDelay defines the amount of time waited between each RPC accepted
+	// from a peer. Without this delay, a peer can force us to spin up thousands
+	// of goroutines per second.
+	peerRPCDelay = build.Select(build.Var{
+		Standard: 1 * time.Second,
+		Dev:      1 * time.Second,
+		Testing:  100 * time.Millisecond,
```
Comment (lukechampine) [inline modules/gateway/consts.go]: 100ms might be too slow for some tests

File: modules/gateway/rpc.go [hunk 2]
```diff
@@ -173,6 +173,10 @@ func (g *Gateway) threadedListenPeer(p *peer) {
 		// The handler is responsible for closing the connection, though a
 		// default deadline has been set.
 		go g.threadedHandleConn(conn)
+
+		if !g.managedSleep(peerRPCDelay) {
```
Comment (DavidVorick) [inline modules/gateway/rpc.go]: This is per-RPC.

There should be a test, yes. Will work on that in a bit.

Comment (DavidVorick) [inline modules/gateway/rpc.go]: * sorry, this ratelimit is per-peer. So one peer cannot drown out the other peers.

We have another ratelimit that allows only one peer to connect per 6 seconds, and that could potentially be drowned out by an adversary. But it only affects inbound connections. One option we thought of was to require a PoW in the handshake.

**Comment by DavidVorick** [comment]
I'm not sure how seriously the new limits will affect the gateway on main-net, so I'm going to be trying it out today. We should also run an antfarm with this update to make sure that peers can still function normally, send transactions, etc.

**Comment by DavidVorick** [comment]
The test I pushed is failing, but seems like more muxado issues and not a result of anything in the gateway code. I've also tested this code in production to confirm that everything works.

Will be good to switch away from muxado asap. I think we'll see a lot of issues disappear upon doing so.

**Comment by lukechampine** [comment]
Is this good to merge then? Or will it cause future builds to fail? If so, we can wrap it in a `t.Skip("skipped until muxado is dropped")` or something.

**Comment by DavidVorick** [comment]
Yeah. Assuming the build passes this should be good to merge.

File: modules/host/consts.go [hunk 3]
```diff
@@ -254,6 +254,21 @@ var (
...
+			return time.Millisecond * 10
+		}
+		if build.Release == "standard" {
+			return time.Millisecond * 50
+		}
+		if build.Release == "testing" {
+			return time.Millisecond
+		}
+		panic("unrecognized release constant in host - obligationLockTimeout")
+	}()
```
Comment (lukechampine) [inline modules/host/consts.go]: `build.Select`?

Comment (DavidVorick) [inline modules/host/consts.go]: This matches all of the other constants in the host. Figured we'd update them all together in a different PR.

**Comment by DavidVorick** [comment]
looks like ndf build failure, but otherwise lgtm

**Pull request merged**